sklearn/datasets/_base.py

```diff
@@ -1109,6 +1110,36 @@ def load_linnerud(*, return_X_y=False, as_frame=False):
     )
 
 
+@deprecated(
+    """`load_boston` is deprecated in 1.0 and will be removed in 1.2.
+
+    The Boston housing prices dataset has an ethical problem. You can refer to
+    the documentation of this function for further details.
+
+    The scikit-learn maintainers therefore strongly discourage the use of this
+    dataset unless the purpose of the code is to study and educate about
+    ethical issues in data science and machine learning.
+
+    In this case special case, you can fetch the dataset from OpenML:
+
+        from sklearn.datasets import fetch_openml
+        boston = fetch_openml('boston', version=2, as_frame=False)
```

Comment:
Unfortunately this is not a drop-in replacement. The `.data` matrix is the same but the `.target` attribute holds values in `{'N', 'P'}` instead of the price information.

Reply:
We need to use `version=1`

Reply:
but `np.testing.assert_allclose` says it is not the same. Let me check if this is just the column order

Reply:
Otherwise it should be possible to use `pandas.read_csv` with clever options to directly read the original:

http://lib.stat.cmu.edu/datasets/boston

Reply:
Got it! The following lines make it possible to reconstruct the same arrays as in scikit-learn from the original text file (I checked):

```python
import pandas as pd
import numpy as np


data_url = "http://lib.stat.cmu.edu/datasets/boston"
df = pd.read_csv(data_url, sep="\s+", skiprows=22, header=None)
data = np.hstack([df.values[::2, :], df.values[1::2, :2]])
target = target = df.values[1::2, 2]
```

Reply:
I think that we can keep the `fetch_openml` indeed.

Reply:
I would rather include the code snippet above instead of relying on `fetch_openml`:

- this is using the original source dataset
- it is a direct drop-in replacement for the numpy arrays returned by `load_boston`.

As you show in https://github.com/scikit-learn/scikit-learn/pull/20729#pullrequestreview-727552983, it's not easy to get the drop-in replacement from `fetch_openml` because of the presence of ordinal encoded categories with a non-zero offset.